src/components/editor/DraftSpec.tsx

```diff
@@ -26,19 +24,18 @@ function DraftSpecEditor() {
     const [draftSpec, setDraftSpec] = useState<DraftSpecQuery | null>(null);
 
     const handlers = {
-        change: (newVal: any) => {
+        change: (newVal: any, catalogName: string) => {
```

Comment:
Makes sense to pass the catalog name so it can better handle async updates (these are not working 100% yet.